File: ui/react/src/action/LayoutAsset.ts
import Layout from "../model/Layout";
import SheetAction from "./SheetAction";
import Action from "./Action";

class LayoutAsset extends SheetAction {
    public assetName: string;
    public layout: Layout;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const assetName: string = input.assetName;
        const layout = Layout.deserialize(input.layout);
        return new LayoutAsset(sheetName, assetName, layout);
    }

    constructor(sheetName: string, assetName: string, layout: Layout) {
        super(sheetName);
        this.assetName = assetName;
        this.layout = layout;
    }

    public isAssetAction(): boolean {
        return true;
    }

    public targetAsset(): string {
        return this.assetName;
    }

    public wrapper(): Action {
        const action = new Action();
        action.layoutAsset = this;
        return action;
    }

}

export default LayoutAsset;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/LayoutAsset.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.api.view.Layout;

public class LayoutAsset extends SheetAction {
    private String assetName;
    private Layout layout;

    public LayoutAsset() {
    }

    public LayoutAsset(String sheetName, String assetName, Layout layout) {
        super(sheetName);
        this.assetName = assetName;
        this.layout = layout;
    }

    public String getAssetName() {
        return assetName;
    }

    public void setAssetName(String assetName) {
        this.assetName = assetName;
    }

    public Layout getLayout() {
        return layout;
    }

    public void setLayout(Layout layout) {
        this.layout = layout;
    }
}
